apis/v1beta1/httproute_types.go

```diff
@@ -889,7 +892,14 @@ type HTTPRequestRedirectFilter struct {
 	// Port is the port to be used in the value of the `Location`
 	// header in the response.
 	//
-	// When empty, the Gateway Listener port is used.
+	// When empty, the redirect port MUST be derived using the following rules:
```

Comment:
Maybe change "When empty" to "If no port is specified", just to be that much more explicit?

This looks good to me, with that minor nit -- thanks!

Reply:
Sounds good. Thanks for the suggestion :)